### ISSUE TITLE
Using hint types for optional arguments

### ISSUE THREAD

**Issue by Pierlou** [pyrates/minicli#12]
Hi, thanks for this lib. We've been using it in [a project](https://github.com/datagouv/hydra/blob/main/udata_hydra/cli.py) and have recently bumped into an unexpected behaviour when hint typing the CLI functions (temporary fix in [this PR](https://github.com/datagouv/hydra/pull/126)). Minimal reproduction (in `test_cli.py`):
```
from typing import Union, Optional
from minicli import cli, run


@cli
def greetings(name: str):
    """This is an example program

    :name: The name of the person we want to greet
    """
    print(f"Hi {name}!")


@cli
def greetings_opt(name: Optional[str] = "John"):
    """This is an example program

    :name: The name of the person we want to greet
    """
    print(f"Hi {name}!")


@cli
def greetings_uni(name: Union[str, None] = "John"):
    """This is an example program

    :name: The name of the person we want to greet
    """
    print(f"Hi {name}!")


if __name__ == '__main__':
    run()
```

Same command for all three: `test_cli.py greetings Jake` (changing the function):
- first function returns `Hi Jake!` (expected)
- both others throw this error: `error: argument --name/-n: invalid Optional value: 'John'`

Is this expected? If so, how should we hint type our optional arguments?

**Comment by Pierlou** [comment]
The issue is on the hint typing part, not so much on the default argument. Specifying the named argument doesn't seem to fix the issue:
```
test_cli.py greetings_opt
```
and
```
test_cli.py greetings_opt Jake
```
both raise `error: argument --name/-n: invalid Optional value: 'John'`, and
```
test_cli.py greetings_opt --name Jake
```
raises `error: argument --name/-n: invalid Optional value: 'Jake'`

**Comment by yohanboniface** [comment]
Ok, sorry that I misunderstood your point.

So to understand better, are you typing your argument for your own usage or are you typing in order to configure the cli behaviour ? 

In other words, I'm wondering if you expect minicli to learn how to parse the typing options (which may or may not be "mini" ;) ), or just skipping the hint when it does not know what to do with it ?

**Comment by yohanboniface** [comment]
I've made a proposal in #13, tell me if this works for you :)

**Comment by Pierlou** [comment]
Thanks for the quick PR! Looks good to me 🙏 (I can't see whether the tests are successful but if they are it fixes the issue for sure). Will you release a new version following this patch?

**Comment by yohanboniface** [comment]
Released 0.5.3 :)
